spec/Staking.md

```diff
@@ -0,0 +1,340 @@
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 123900, 100],
+    ["0xAB..CD", "0x01..23", 124000, 50],
+  ]
+}
+
+> "0x23..45" sends staking action [3, "0x01..23", 500"] at block timestamp 123600
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 123900, 100],
+    ["0xAB..CD", "0x01..23", 124000, 50],
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 123900
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 790,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 100], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 124000, 50],
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 124000
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 840,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 50], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 124100
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0x23..45": 500,
+    "0xAB..CD": 840,
+  },
+  delegation: {
+    "0xAB..CD": [ ["0x01..23", 50], ["0x23..45", 110] ]
+  }
+}
+```
+
+# Fee distribution
+
+You pay fees to make a transaction, which is greater than a specified minimum fee for the transaction type. Transaction fees within a block are collected and CCS holders and the block author share this.
+
+CCS Holders share the sum of the minimum fees for the transaction. They get fees in proportion to the sum of the balance of CCS plus the total amounts of CCS that are delegated to someone.
+
+Fees distributed to CCS holders will be rounded down to an integer. The block author gets the rest, which is the sum of the remaining amount due to rounding and the amount that exceeds the minimum fee.
+
+## Example
+
+```
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0x23..45": 450,
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 50] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 124000, 50],
+  }
+}
+```
+
+In a such state, an account "0x23..45" has a weight of 500 (balance 450 + delegation 50), and "0xAB..CD" has a weight of 1000 (balance 690 + delegation 200 + delegation 110, pending revocation will not affect to it). If a block has total fees of 110, and if the sum of the minimum fees for it is 35, each account will share the following amount of fees.
```

Comment:
In a such state, an account "0x23..45" has a weight of 500 (balance 450 + delegation 50), and "0xAB..CD" has a weight of 1000 (balance 690 + delegation 200 + delegation 110, pending revocation will not affect it). If a block has total fees of 110, and if the sum of the minimum fees for it is 35, each account will share the following amount of fees.